core/src/main/scala/org/http4s/instances/FunctionKInstances.scala

```diff
@@ -0,0 +1,12 @@
+package org.http4s.instances
+
+import cats.Functor
+import cats.arrow.FunctionK
+import cats.data.OptionT
+
+trait FunctionKInstances {
```

Comment:
I am a little reluctant to provide this when it's 100% cats functionality.  Have we talked to the Cats project about whether this would be a welcome function there?

Reply:
I implemented this with an extension method on `Kleisli` instead, so instead of
```scala
kleisliOfF.transform(FToOptionT)
```
it's now
```scala
kleisliOfF.liftOptionT
```